spec/support/shared_examples/mailer_preview.rb

```diff
@@ -0,0 +1,27 @@
+RSpec.shared_examples 'a mailer preview' do
+  let(:mailer_class) { described_class.class_name.gsub(/Preview$/, '').constantize }
+
+  it 'has a preview method for each mailer method' do
+    mailer_methods = mailer_class.instance_methods(false)
+    preview_methods = described_class.instance_methods(false)
+    expect(mailer_methods - preview_methods).to be_empty
+  end
+
+  described_class.instance_methods(false).each do |mailer_method|
```

Comment:
honestly I am shocked that `described_class` works at the class level like this!